test/npu-xrt/packet_flow/aie.mlir

```diff
@@ -0,0 +1,132 @@
+//===- aie.mlir ------------------------------------------------*- MLIR -*-===//
+//
+// This file is licensed under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+// (c) Copyright 2022-2024 Advanced Micro Devices, Inc. or its affiliates
+// Copyright (C) 2020-2022, Xilinx Inc.
+//
+//===----------------------------------------------------------------------===//
+
+module {
+  aie.device(npu1_1col) {
+    memref.global "public" @objFifo_in0 : memref<56x56xi8>
+    memref.global "public" @objFifo_out0 : memref<64x64xi8>
+
+    %tile_0_0 = aie.tile(0, 0)
+    %tile_0_1 = aie.tile(0, 1)
+    %tile_0_2 = aie.tile(0, 2)
+
+    %objFifo_in1_cons_buff_0 = aie.buffer(%tile_0_2) {sym_name = "objFifo_in1_cons_buff_0"} : memref<64x64xi8>
+    %objFifo_in1_cons_buff_1 = aie.buffer(%tile_0_2) {sym_name = "objFifo_in1_cons_buff_1"} : memref<64x64xi8>
+    %objFifo_out1_buff_0 = aie.buffer(%tile_0_2) {sym_name = "objFifo_out1_buff_0"} : memref<64x64xi8>
+    %objFifo_out1_buff_1 = aie.buffer(%tile_0_2) {sym_name = "objFifo_out1_buff_1"} : memref<64x64xi8>
+
+    %objFifo_in1_cons_prod_lock = aie.lock(%tile_0_2, 0) {init = 1 : i32, sym_name = "objFifo_in1_cons_prod_lock"}
+    %objFifo_in1_cons_cons_lock = aie.lock(%tile_0_2, 1) {init = 0 : i32, sym_name = "objFifo_in1_cons_cons_lock"}
+    %objFifo_out1_prod_lock = aie.lock(%tile_0_2, 2) {init = 1 : i32, sym_name = "objFifo_out1_prod_lock"}
+    %objFifo_out1_cons_lock = aie.lock(%tile_0_2, 3) {init = 0 : i32, sym_name = "objFifo_out1_cons_lock"}
+
+    aie.packet_flow(0) {
+      aie.packet_source<%tile_0_1, DMA : 0>
+      aie.packet_dest<%tile_0_2, DMA : 0>
+    }
+    aie.packet_flow(1) {
+      aie.packet_source<%tile_0_2, DMA : 0>
+      aie.packet_dest<%tile_0_1, DMA : 1>
+    }
+    aie.packet_flow(2) {
+      aie.packet_source<%tile_0_1, DMA : 1>
+      aie.packet_dest<%tile_0_0, DMA : 0>
+    }
+
+    // TODO: Replace below with packet flow once we have shim dma op issuing packet headers.
+    aie.flow(%tile_0_0, DMA : 0, %tile_0_1, DMA : 0)
```

Comment:
You can use packet mode on the shim dma but you need to use `npu.writebd` or `npu.blockwrite` to write the BD. See for example #1666

Reply:
Totally agreed. Thanks for pointing to the code example. I was wondering if it would be convenient to make npu.dma_memcpy_nd op able to issue packet by using this PR: https://github.com/Xilinx/mlir-aie/pull/1671

Then we can convert this PR to be the board test for the new packet header attribute.